### DISCUSSION TITLE
Gradient with respect to a tensor created internally

### DISCUSSION THREAD

**Discussion by davisyoshida** [jax-ml/jax#5819] [Q&A] [answered]
I have a 18 layer network, and need the gradient of the output with respect to some of the tensors created during the forward pass. Is there a recommended way to go about doing this? I couldn't figure out a good solution with `value_and_grad`.

**Comment by mattjj** [comment, marked as answer]
Thanks for the question! I think [this answer](https://github.com/google/jax/discussions/5336#discussioncomment-269983) might cover what you have in mind.

**Reply by davisyoshida** [reply]
That sounds perfect. Thanks! Sorry for the bother.

**Reply by mattjj** [reply]
No bother; we like questions!
